doc/flamegraphs.md

```diff
@@ -0,0 +1,176 @@
+that can create stack data in the correct format can be used to generate a flame
+graph. The flame graphs themselves are SVG files that have a small have a small
+amount of embedded Javascript that provide mouseover events and some functions
+for zooming and filtering. They can also be viewed in regular image viewers or
+browsers that have Javascript enabled.
+
+**TODO:** Try to get this hosted on bitcoincore.org, see [this
+PR](https://github.com/bitcoin-core/bitcoincore.org/pull/522).
+
+Example (**[interactive SVG](https://monad.io/bitcoin-flamegraph.svg)**):
+
+![Example Flamegraph](https://monad.io/bitcoin-flamegraph.svg)
+
+You can learn more about flame graphs
+[on Brendan Gregg's site](http://www.brendangregg.com/flamegraphs.html).
+
+### Interpreting Flame Graphs
+
+Height on the flame graph represents different stack traces, and width
+represents number of samples (i.e. percentage of total time). Tall sections are
+for deep stacks, and wide sections represent parts of the programming take a lot
+of time.
+
+The flame graph is color coded according to certain heuristics:
+
+ * **Red** frames are attributed to libc or system libraries
+ * **Yellow** frames are attributed to third party C++ libraries
+ * **Green** frames are attributed to Bitcoin source files
+ * **Blue** frames are attributed to the Linux kernel
+
+## Installing The Flame Graph Scripts
+
+The upstream GitHub repo for flame graphs can be found at
+[github.com/brendangregg/FlameGraph](https://github.com/brendangregg/FlameGraph).
+To run the scripts you'll need Perl installed on your computer.
+
+There is a small optional patch in the Bitcoin repository at
+[contrib/devtools/flamegraph.patch](/contrib/devtools/flamegraph.patch). You can
+apply this patch against the upstream FlameGraph `v1.0` tag to generate flame
+graphs that are color coded for Bitcoin. You can apply the patch using a command
+like:
+
+```bash
+# Assuming you're at the top directory in the FlameGraph repo.
+$ patch -p1 <path/to/contrib/devtools/flamegraph.patch
+```
+
+## Generating Flame Graphs On Linux
+
+The Linux kernel project maintains a command called `perf` that provides a high
+level interface to the
+[perf_event_open(2)](http://www.man7.org/linux/man-pages/man2/perf_event_open.2.html)
+system call. Perf lets you generate accurate CPU profiles of kernel events and
+userspace applications with relatively low overhead. This is the recommended way
+to generate flame graphs on Linux, as it is lower overhead than other options
+and allows generating profiles that can inspect kernel time.
+
+For more information about `perf`:
+
+ * [Perf Tutorial](https://perf.wiki.kernel.org/index.php/Tutorial)
+ * [Perf Examples](http://www.brendangregg.com/perf.html)
+
+### Setting Up Perf
+
+Perf is available on most Linux distros in a package named `perf` or some
+variant thereof:
+
+```bash
+# Install perf on Debian.
+sudo apt-get install linux-perf
```

Comment:
Seems on Ubuntu you need `linux-tools-generic`, `linux-cloud-tools-generic`, `linux-tools-common`.